gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/RpcBatch.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns.spi;
+
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.ManagedZonesListResponse;
+import com.google.api.services.dns.model.Project;
+
+import java.util.Map;
+
+/**
+ * An interface for the collection of batch operations.
+ */
+public interface RpcBatch {
+
+  /**
+   * Adds a call to "list zones" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addListZones(DnsRpc.Callback<ManagedZonesListResponse> callback,
```

Comment:
I am fine with keeping the Callback interface as part of `DnsRpc` however because it is only used in this context
you may want to consider moving it to here.